bevy_widgets/bevy_numeric_field/examples/transform.rs

```diff
@@ -0,0 +1,212 @@
+    });
+    let scale_z = spawn_field(&mut commands, |val, transform| {
+        transform.scale.z = val;
+    });
+
+    let rotation_x = spawn_field(&mut commands, |val, transform| {
+        transform.rotation.x = val;
+    });
+    let rotation_y = spawn_field(&mut commands, |val, transform| {
+        transform.rotation.y = val;
+    });
+    let rotation_z = spawn_field(&mut commands, |val, transform| {
+        transform.rotation.z = val;
+    });
+    let rotation_w = spawn_field(&mut commands, |val, transform| {
+        transform.rotation.w = val;
+    });
+
+    let transform_widget = TransformWidget {
+        transform: [translate_x, translate_y, translate_z],
+        scale: [scale_x, scale_y, scale_z],
+        rotation: [rotation_x, rotation_y, rotation_z, rotation_w],
+        target,
+    };
+
+    let root = commands
+        .spawn(NodeBundle {
+            style: Style {
+                width: Val::Percent(25.0),
+                height: Val::Percent(100.0),
+                display: Display::Grid,
+                justify_content: JustifyContent::Center,
+                align_items: AlignItems::Center,
+                grid_template_columns: vec![GridTrack::min_content(), GridTrack::flex(1.0)],
+                grid_template_rows: vec![GridTrack::min_content(); 13],
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .id();
+
+    header_text(&mut commands, root, "Translate");
+
+    child_text(&mut commands, root, "X");
+    commands.entity(root).add_child(translate_x);
+    child_text(&mut commands, root, "Y");
+    commands.entity(root).add_child(translate_y);
+    child_text(&mut commands, root, "Z");
+    commands.entity(root).add_child(translate_z);
+
+    header_text(&mut commands, root, "Scale");
+
+    child_text(&mut commands, root, "X");
+    commands.entity(root).add_child(scale_x);
+    child_text(&mut commands, root, "Y");
+    commands.entity(root).add_child(scale_y);
+    child_text(&mut commands, root, "Z");
+    commands.entity(root).add_child(scale_z);
+
+    header_text(&mut commands, root, "Rotation");
+
+    child_text(&mut commands, root, "X");
+    commands.entity(root).add_child(rotation_x);
+    child_text(&mut commands, root, "Y");
+    commands.entity(root).add_child(rotation_y);
+    child_text(&mut commands, root, "Z");
+    commands.entity(root).add_child(rotation_z);
+    child_text(&mut commands, root, "W");
+    commands.entity(root).add_child(rotation_w);
+
+    commands.insert_resource(transform_widget);
+}
+
+fn spawn_field(
+    commands: &mut Commands,
+    callback: impl Fn(f32, &mut Transform) + Send + Sync + 'static,
+) -> Entity {
+    commands
+        .spawn((
+            NodeBundle {
+                style: Style {
+                    width: Val::Px(300.0),
+                    height: Val::Px(30.0),
+                    margin: UiRect::all(Val::Px(5.0)),
+                    ..default()
```

Comment:
same as above comment.
maybe just check everywhere.
is there a lint we can use for this